File: Exerc/matriz_quadrada_2/src/Main.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.io.PrintWriter;

public class Main {
 
    public static void main(String[] args) throws IOException {
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(System.out);
        String linha;
        
        while ((linha = br.readLine()) != null && !linha.isEmpty()) {
            int N = Integer.parseInt(linha);
            
            if (N == 0) {
                break;
            }
            
            int[][] M = new int[N][N];
            
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    M[i][j] = Math.abs(i - j) + 1;
                }
            }
            
            StringBuilder sb = new StringBuilder();
            
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    // Garante a formatação e o alinhamento
                    if (M[i][j] < 10) {
                        sb.append("  ").append(M[i][j]);
                    } else if (M[i][j] < 100) {
                        sb.append(" ").append(M[i][j]);
                    } else {
                        sb.append(M[i][j]);
                    }
                    
                    if (j < N - 1) {
                        sb.append(" ");
                    }
                }
                sb.append("\n");
            }
            sb.append("\n");
            pw.print(sb.toString());
        }
        
        pw.flush();
    }
}

File: Exerc/matriz_quadrada_2/bin/Main.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let lineIndex = 0;
while (true) {
    let N = parseInt(lines[lineIndex++]);
    
    if (isNaN(N) || N === 0) {
        break;
    }
    
    let matriz = [];
    for (let i = 0; i < N; i++) {
        matriz[i] = [];
        for (let j = 0; j < N; j++) {
            matriz[i][j] = Math.abs(i - j) + 1;
        }
    }
    
    // Otimização na impressão
    let resultado = [];
    for (let i = 0; i < N; i++) {
        let linha = [];
        for (let j = 0; j < N; j++) {
            let num = matriz[i][j];
            // Formatação de alinhamento
            if (num < 10) {
                linha.push("  " + num);
            } else if (num < 100) {
                linha.push(" " + num);
            } else {
                linha.push(num);
            }
        }
        resultado.push(linha.join(" "));
    }
    
    console.log(resultado.join("\n") + "\n");
}